### PR TITLE
Update VSP/ticket buying pages.

### PR DESCRIPTION
**Pull request by jholdstock**
Reviewed most of the VSP/ticket buying pages to ensure they are up to date. Most of the detail is in individual commit messages.

- Removed most references to Redeem Scripts. What remains is on a dedicated page which is clearly labelled as outdated information (`redeem-script.md`)
- Removed "legacy" staking instructions. Simplified the remaining staking instructions by removing duplicated and overly verbose information.

### CODE REVIEW

**Comment by jholdstock** [comment]
Tagging some potential reviewers - @matthawkins90 @xaur

File: docs/wallets/cli/dcrwallet-tickets.md [hunk 1]
```diff
@@ -54,148 +50,63 @@ Once `dcrwallet` is restarted with that line in `dcrwallet.conf`, your wallet wi
...
+; The base64 encoded public key of the VSP server.  This can be found on the
+; VSP website in the footer.
+vsp.pubkey=ia9Ra2Drb+OHLqRyBsJnRKBd7TUG1IvrseC6robKzGo=
 ```
 
-1. `fromaccount`    =  Required String: The account from which to purchase tickets (e.g. "default").
-1. `spendlimit`     =  Required Number: Limit on the amount to spend on ticket (e.g. 50).
-1. `minconf`        =  Optional Number: Minimum number of block confirmations required (e.g. 1).
-1. `ticketaddress`  =  Optional String: The ticket address to which voting rights are given
-1. `numtickets`     =  Optional Number: The number of tickets to purchase at once (e.g. 1)
```
Comment (xaur) [inline docs/wallets/cli/dcrwallet-tickets.md:118]: `numtickets` is still useful for purchasing multiple tickets at once. But to use it you have to also satisfy `minconf` and `ticketaddress` (emtpy string for VSPv3) params AND it doesn't always work as expected (see other comments) so not sure if we want to mention it here, omit, or a middleground "run `dcrctl --wallet help purchaseticket` for advanced options".

**Review by xaur — COMMENTED** [review]
Thanks for refreshing this!

In addition to a few line comments, possible changes for this or future PRs (to not drag this PR for too long):

- My experience with dcrctl+dcrwallet v1.7.x has been rough at times. `purchaseticket` does not always purchase the requested `numtickets` amount of tickets, a ton of funds tend to get locked as `unconfirmed` in additional split transactions (both for sizing ticket purchases and VSP fees), and the whole cycle takes much longer than it was with dcrstakepool. It may be worth mentioning of what to expect, i.e. get ready to have your wallet running and unlocked for hours to buy multiple tickets.
- I'd mention for how long after `purchaseticket` it is recommended to NOT shutdown dcrwallet (e.g. at the very least until all fees have been sent to the VSP - but then we must tell the user how to know from the log that it happened).
- Maybe also mention that the wallet (or account) must remain unlocked to allow communication with the VSP, up until all VSP fees are confirmed.
- May be worth mentioning the new `processunmanaged` command that can assign tickets to the configured VSP if those tickets have not been assigned for whatever reason (e.g. you accidentally purchases tickets BEFORE configuring the VSP and nobody is monitoring them).
- It is possible to use multiple VSPs at the same time, but you have to edit the config (or switch config files) before every dcrwallet launch for your ticket purchasing session.

General:

- Nice to still have a page for redeem scripts. It may help a few users who haven't checked Decred for years to recover their stuff.
- Appreciate individual commits, easy to review (esp. having removals separately).

File: docs/proof-of-stake/how-to-stake.md [hunk 2]
```diff
@@ -2,28 +2,44 @@
...
-## How to Stake/Vote
+Participating in Decred’s Proof-of-Stake system requires the use of an
+always-online voting wallet.
+This wallet must remain online and unlocked 24/7, ready to sign vote
+transactions immediately when tickets are selected to vote on a block.
 
-There are two ways to stake/vote: solo staking and staking using a VSP. The former method is usually more appropriate for advanced users as it requires your wallet to be up 24/7. Running your own always-online wallet is known as **"Solo" Voting**.
-
-**Voting Service Providers (VSPs)** (formerly "Stakepools") are available for those unable to keep a personal voting wallet online. Using a VSP is completely safe; the Decred PoS protocol allows you to delegate your vote to a VSP's always-online wallet without ever giving the VSP access to your funds.
+Advanced users with some sys-admin skills may choose to run their own voting wallet.
```
[the remark this answers is not in the document]
Comment (jholdstock) [reply docs/proof-of-stake/how-to-stake.md:10]: Removed the hyphen

File: docs/wallets/cli/dcrwallet-tickets.md (hunk 1 above)
Comment (jholdstock) [reply docs/wallets/cli/dcrwallet-tickets.md:118]: Added the help command so advanced users can explore it themselves.

File: docs/glossary.md [hunk 3]
```diff
@@ -388,6 +388,15 @@ The minimum level of participation required in order for a decision-making proce
...
 ---
 
+#### Redeem Script
+
+A [redeem script](proof-of-stake/redeem-script.md) is 1-of-2 multisignature
+script generated during the user registration process of
+[dcrstakepool](https://github.com/decred/dcrstakepool) (legacy VSP software
+superseded by [vspd](https://github.com/decred/vspd)).
+Allocating voting rights to the hash of this shared script would enable either
+the user or the VSP to vote tickets purchased by the user.
```
[the remark this answers is not in the document]
Comment (jholdstock) [reply docs/glossary.md:398]: Added

**Comment by jholdstock** [comment]
Thanks for the review.

Implemented some of your suggestions, but as you mention, the rest are probably good for another PR or maybe even another page? Things like using multiple VSPs or registering historic tickets with VSPs are more advanced imo, not suitable for the beginners guide. The main goal of this PR was to just remove the completely outdated info.

**Comment by xaur** [comment]
Agreed, this is already a nice cleanup/update and good to go as is. Better move in smaller steps but get more stuff merged.

Appreciate the example log lines, now users will know exactly what to look for.

For future PRs I'd suggest keep it on the same page but move `processunmanaged`, multi-VSP and other tricks to "Advanced..." section at the end. The only thing I'd keep in the main section is the `numtickets` parameter, it looks like it should be pretty common.

File: docs/wallets/cli/dcrwallet-tickets.md [hunk 4]
```diff
@@ -70,79 +65,46 @@ vsp.pubkey=ia9Ra2Drb+OHLqRyBsJnRKBd7TUG1IvrseC6robKzGo=
 
 ---
 
-## Ticket Purchasing
+## Step 2. Purchase Tickets
+
+Tickets can be purchased in two different ways, manual or automatic. Purchasing
```
Comment (matthawkins90) [inline docs/wallets/cli/dcrwallet-tickets.md:70]: Isn't it true that manual ticket buying will never be mixed, but automatic ticket buying CAN be mixed if you've configured with cspp? Or is automatic ticket buying ALWAYS mixed via cspp?

Either way, I think those details should be briefly mentioned here.

**Review by matthawkins90 — CHANGES_REQUESTED** [review]
@jholdstock, I'm super sorry it took me 3 weeks to review this.
Overall, great work. Appreciate you doing all this cleanup.

File: docs/proof-of-stake/how-to-stake.md [hunk 5]
```diff
@@ -2,26 +2,44 @@
...
+This is known as [Solo voting](#solo-pos-voting).
+Those who are unable or unwilling to maintain an always-online voting
```
[the remark this answers is not in the document]
Comment (jholdstock) [reply docs/proof-of-stake/how-to-stake.md:12]: Removing the overly soft/conversational language was deliberate, an attempt to match the style in the rest of the site. Looking back at this I can see that I've probably taken it too far in the other direction. I'll add your suggestion.

For some background - the conversational style stems from the fact that some of the docs are literally conversations which were copy/pasted from 2015-16 forum threads.

File: docs/wallets/cli/dcrwallet-tickets.md (hunk 4 above)
Comment (jholdstock) [reply docs/wallets/cli/dcrwallet-tickets.md:70]: `how-to-cspp.md` explains how to set up mixed ticket buying. I will add a link.

**Review by dajohi — APPROVED** [review]

File: docs/wallets/cli/dcrwallet-tickets.md (hunk 4 above)
Comment (xaur) [reply docs/wallets/cli/dcrwallet-tickets.md:70]: I also don't know if manual (`dcrctl --wallet purchaseticket`) can mix but the autobuyer can be used *without* mixing.

**Pull request merged**